packages/persist/lib/records.ts

```diff
@@ -0,0 +1,143 @@
+import { records as recordsService, format as recordsFormatter } from '@nangohq/records';
+import type { FormattedRecord, UnencryptedRecordData, UpsertSummary } from '@nangohq/records';
+import { errorManager, ErrorSourceEnum, LogActionEnum, updateSyncJobResult, getSyncConfigByJobId } from '@nangohq/shared';
+import tracer from 'dd-trace';
+import type { Span } from 'dd-trace';
+import { logContextGetter } from '@nangohq/logs';
+import type { Result } from '@nangohq/utils';
+import { Err, Ok, metrics, stringifyError } from '@nangohq/utils';
+
+export type PersistType = 'save' | 'delete' | 'update';
+export const recordsPath = '/environment/:environmentId/connection/:nangoConnectionId/sync/:syncId/job/:syncJobId/records';
+
+export async function persistRecords({
+    persistType,
+    environmentId,
+    connectionId,
+    providerConfigKey,
+    nangoConnectionId,
+    syncId,
+    syncJobId,
+    model,
+    records,
+    activityLogId
+}: {
+    persistType: PersistType;
+    environmentId: number;
+    connectionId: string;
+    providerConfigKey: string;
+    nangoConnectionId: number;
+    syncId: string;
+    syncJobId: number;
+    model: string;
+    records: Record<string, any>[];
+    activityLogId: string;
+}): Promise<Result<void>> {
+    const active = tracer.scope().active();
+    const recordsSizeInBytes = Buffer.byteLength(JSON.stringify(records), 'utf8');
+    const span = tracer.startSpan('persistRecords', {
+        childOf: active as Span,
+        tags: {
+            persistType,
+            environmentId,
+            connectionId,
+            providerConfigKey,
+            nangoConnectionId,
+            syncId,
+            syncJobId,
+            model,
+            activityLogId,
+            'records.count': records.length,
+            'records.sizeInBytes': recordsSizeInBytes
+        }
+    });
+
+    let persistFunction: (records: FormattedRecord[]) => Promise<Result<UpsertSummary>>;
+    let softDelete: boolean;
+    switch (persistType) {
+        case 'save':
+            softDelete = false;
+            persistFunction = async (records: FormattedRecord[]) => recordsService.upsert({ records, connectionId: nangoConnectionId, model, softDelete });
+            break;
+        case 'delete':
+            softDelete = true;
+            persistFunction = async (records: FormattedRecord[]) => recordsService.upsert({ records, connectionId: nangoConnectionId, model, softDelete });
+            break;
+        case 'update':
+            softDelete = false;
+            persistFunction = async (records: FormattedRecord[]) => {
+                return recordsService.update({ records, connectionId: nangoConnectionId, model });
+            };
+            break;
+    }
```

Comment:
this function is exactly the same except this switch logic which is setting the persistFunction based on the persistType instead of passing both the type and the function as arguments